core/src/androidMain/kotlin/com/revenuecat/purchases/kmp/Purchases.android.kt

```diff
@@ -67,10 +69,13 @@ public actual class Purchases private constructor(private val androidPurchases:
             with(configuration) {
                 // Using the common configure() call allows us to pass PlatformInfo.
                 commonConfigure(
-                    AndroidProvider.requireApplication(),
+                    context = AndroidProvider.requireApplication(),
                     apiKey = apiKey,
                     appUserID = appUserId,
-                    observerMode = observerMode,
+                    observerMode = when (configuration.purchasesAreCompletedBy) {
+                        REVENUECAT -> false
+                        MY_APP -> true
```

Comment:
We could potentially have this enum as an internal property of the enum, so we don't duplicate the logic for iOS and android. Having said that, that logic would disappear once we update to latest iOS/android, so NABD.

Reply:
I like this! (We have the same in Android.) But I went to implement it and got: 
>`when' with expect enum as subject can not be exhaustive without else branch  
  
It can be exhaustive if we do it in the platform source sets, but then we're duplicating again 😅 

So I'd say it's prolly not worth it. Also serves as a good visual indicator that we need to remove this once PHC depends on the latest iOS/Android. 😛